crates/wasmtime/src/module/serialization.rs

```diff
@@ -0,0 +1,715 @@
+            "dynamic memory guard size",
+        )?;
+        Self::check_bool(
+            self.tunables.generate_native_debuginfo,
+            other.generate_native_debuginfo,
+            "debug information support",
+        )?;
+        Self::check_bool(
+            self.tunables.parse_wasm_debuginfo,
+            other.parse_wasm_debuginfo,
+            "WebAssembly backtrace support",
+        )?;
+        Self::check_bool(
+            self.tunables.interruptable,
+            other.interruptable,
+            "interruption support",
+        )?;
+        Self::check_bool(
+            self.tunables.consume_fuel,
+            other.consume_fuel,
+            "fuel support",
+        )?;
+        Self::check_bool(
+            self.tunables.static_memory_bound_is_maximum,
+            other.static_memory_bound_is_maximum,
+            "pooling allocation support",
+        )?;
+
+        // At this point, the hashes should match (if not we're missing a check)
+        assert_eq!(
+            Self::simple_hash(&self.tunables),
+            Self::simple_hash(other),
+            "unexpected hash difference"
+        );
+
+        Ok(())
+    }
+
+    fn check_features(&self, compiler: &Compiler) -> Result<()> {
+        let other = compiler.features();
+        Self::check_bool(
+            self.features.reference_types,
+            other.reference_types,
+            "WebAssembly reference types support",
+        )?;
+        Self::check_bool(
+            self.features.multi_value,
+            other.multi_value,
+            "WebAssembly multi-value support",
+        )?;
+        Self::check_bool(
+            self.features.bulk_memory,
+            other.bulk_memory,
+            "WebAssembly bulk memory support",
+        )?;
+        Self::check_bool(
+            self.features.module_linking,
+            other.module_linking,
+            "WebAssembly module linking support",
+        )?;
+        Self::check_bool(self.features.simd, other.simd, "WebAssembly SIMD support")?;
+        Self::check_bool(
+            self.features.threads,
+            other.threads,
+            "WebAssembly threads support",
+        )?;
+        Self::check_bool(
+            self.features.tail_call,
+            other.tail_call,
+            "WebAssembly tail-call support",
+        )?;
+        Self::check_bool(
+            self.features.deterministic_only,
+            other.deterministic_only,
+            "WebAssembly deterministic-only support",
+        )?;
+        Self::check_bool(
+            self.features.multi_memory,
+            other.multi_memory,
+            "WebAssembly multi-memory support",
+        )?;
+        Self::check_bool(
+            self.features.exceptions,
+            other.exceptions,
+            "WebAssembly exceptions support",
+        )?;
+        Self::check_bool(
+            self.features.memory64,
+            other.memory64,
+            "WebAssembly 64-bit memory support",
+        )?;
```

Comment:
Similar to above using a destructuring statement to exhaustively list all fields I think might work well here too?

Reply:
Definitely will add 👍